### PR TITLE
New mapping helper: Spaceproof/Group directional windows, new maint room

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* New mapping helper: Spaceproof directional windows
  * If this helper is placed on a turf, it will check nearby turfs, and makes a directional window on each turf.
  * If there's a door or an airlock on a nearby turf, the center turf will be changed into a floor tile. This is because putting a directional window on an airlock isn't a good idea.
  * Putting it on a normal turf won't work it well.
* Group directional windows
  * This will put inner-directional windows within its chained group.
* New maint room
* A new turf proc to get adjacent turf by given direction
* Made a list into multiple lines instead of a single line - It's hard to recognise which one is included.

## Why It's Good For The Game
Eh... I don't know. I just made it for fun.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

</details>

## Changelog
:cl:
add: a new mapping helper - Spaceproof directional windows, Group directional windows
add: a new 10x10 maint room. You can stare at space as much as you want... for some reason there are items stuck in space??
add: a new turf proc /turf/proc/get_nearby_turf_by_dir(direction). It returns a turf from the turf that is next to from the given direction.
code: minor change to map template list code
/:cl:

### CODE REVIEW

**Comment by BriggsIDP** [comment]
Would be very nice to have the a opposite helper too that places directional windows on the inside of the tile helper is on.

**Comment by EvilDragonfiend** [comment]
> Would be very nice to have the a opposite helper too that places directional windows on the inside of the tile helper is on.

Where would you need that placer?

**Comment by BriggsIDP** [comment]
> Where would you need that placer?

Places like the fland bridge garden area. It's become quite common to make these now

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -537,3 +537,26 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
 			var/obj/O = each
 			for(var/mob/M in O.contents)
 				. += M
+
+/turf/proc/get_nearby_turf_by_dir(target_dir)
```
Comment (itsmeow) [inline code/game/turfs/turf.dm]: this is what `get_step`/`step` does.

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -331,3 +331,73 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+	. = ..()
+	if(!z || !x || !y)
+		CRASH("It's not unable to place directional windows - xyz is null.")
+
+	var/turf/my_turf = get_turf(src)
+	var/list/nearby_turfs = list(
+		my_turf.get_nearby_turf_by_dir(SOUTH),
+		my_turf.get_nearby_turf_by_dir(NORTH),
+		my_turf.get_nearby_turf_by_dir(WEST),
+		my_turf.get_nearby_turf_by_dir(EAST))
```
Comment (itsmeow) [inline code/modules/mapping/mapping_helpers.dm]: This is also not necessary, you can use get_adjacent_turfs or iterate over directions. There are already helpers to retrieve a list of turfs in each direction.

**Review by itsmeow — DISMISSED** [review]

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (EvilDragonfiend) [reply code/modules/mapping/mapping_helpers.dm]: the correct proc name was `get_adjacent_open_turfs` for that. now it makes sense, thanks.

**Comment by EvilDragonfiend** [comment]
image
image
image

* Code is improved
* Group directional window (inner) mapping helper is added

**Review by PestoVerde322 — APPROVED** [review]
More helpers good

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**